infra/bisector.py

```diff
@@ -113,40 +111,52 @@ def bisect(commit_old, commit_new, testcase, fuzz_target, build_data):
 
   Returns:
     The commit SHA that introduced the error or None
+
+  Raises:
+    ValueError: when a repo url can't be determine from the project
   """
-  local_store_path = tempfile.mkdtemp()
-  repo_url = build_specified_commit.infer_main_repo(build_data.project_name,
-                                                    local_store_path,
-                                                    commit_old)
-  bisect_repo_manager = repo_manager.RepoManager(repo_url, local_store_path)
-  commit_list = bisect_repo_manager.get_commit_list(commit_old, commit_new)
-  build_specified_commit.build_fuzzer_from_commit(
-      build_data.project_name, commit_list[0], bisect_repo_manager.repo_dir,
-      build_data.engine, build_data.sanitizer, build_data.architecture,
-      bisect_repo_manager)
-  error_code = helper.reproduce_impl(build_data.project_name, fuzz_target,
-                                     False, [], [], testcase)
-  old_idx = len(commit_list) - 1
-  new_idx = 0
-  if len(commit_list) == 1:
-    if not error_code:
-      return None
-    return commit_list[0]
-
-  while old_idx - new_idx != 1:
-    curr_idx = (old_idx + new_idx) // 2
+  with tempfile.TemporaryDirectory() as tmp_dir:
+    repo_url, repo_name = build_specified_commit.detect_main_repo_from_docker(
+        build_data.project_name, commit_old)
+    if not repo_url or not repo_name:
+      raise ValueError('Main git repo can not be determined.')
+
+    bisect_repo_manager = repo_manager.RepoManager(
+        repo_url, tmp_dir, repo_name=repo_name)
+    commit_list = bisect_repo_manager.get_commit_list(commit_old, commit_new)
     build_specified_commit.build_fuzzer_from_commit(
-        build_data.project_name, commit_list[curr_idx],
-        bisect_repo_manager.repo_dir, build_data.engine, build_data.sanitizer,
-        build_data.architecture, bisect_repo_manager)
-    error_exists = (
-        helper.reproduce_impl(build_data.project_name, fuzz_target, False, [],
-                              [], testcase) == error_code)
-    if error_exists == error_code:
-      new_idx = curr_idx
-    else:
-      old_idx = curr_idx
-  return commit_list[new_idx]
+        build_data.project_name, commit_list[0], bisect_repo_manager.repo_dir,
+        build_data.engine, build_data.sanitizer, build_data.architecture,
+        bisect_repo_manager)
+    initial_error_code = helper.reproduce_impl(build_data.project_name,
```

Comment:
maybe rename this to `current_error_code` or `expected_error_code`. "initial" is a bit confusing, given that it's actually from the newest commit in the range

Reply:
Done.